viewer/packages/pointclouds/src/picking.ts

```diff
@@ -40,24 +40,34 @@ export function intersectPointClouds(
   input: IntersectInput,
   threshold: number = 0.05 // 5 cm
 ): IntersectPointCloudNodeResult[] {
-  const { normalizedCoords, camera } = input;
+  const { normalizedCoords, camera, renderer } = input;
   normalized.set(normalizedCoords.x, normalizedCoords.y);
   raycaster.setFromCamera(normalizedCoords, camera);
   raycaster.params.Points = { threshold };
 
-  const intersections = raycaster.intersectObjects(nodes, true);
+  const intersections: PickPoint[] = [];
+
+  nodes.forEach(node => {
+    const intersection = node.pick(renderer, camera, raycaster.ray);
+
+    if (intersection === null) return;
+
+    intersections.push(intersection);
+  });
+
   return intersections
-    .filter(x => isPointAcceptedByClippingPlanes(x.point, input.clippingPlanes))
+    .sort((x, y) => x.position!.distanceTo(camera.position) - y.position!.distanceTo(camera.position))
```

Comment:
It's because of how it is implemented in potree package but we can make it required because I assume it doesn't make sense to have a point without defined position.

Reply:
Made it required